src/js/TreeView.js

```diff
@@ -7,6 +7,9 @@ function TreeViewItem(label) {
 
 	this.element = document.createElement('div');
 
+	// Store reference to itself. Needed for delegated mouseover
+	this.element.item = this;
```

Comment:
Sounds like a neat solution without adding any non-standard attribute to a DOM element. I'll look into this as soon as I find the time!